src/mission_companion.cpp

```diff
@@ -1668,47 +1667,52 @@ void talk_function::field_harvest( npc &p, const std::string &place )
                 if( tmp.typeId() == plant_types[plant_index] ) {
                     number_plots++;
 
-                    int plant_count = rng( skillLevel / 2, skillLevel );
-                    plant_count *= bay.furn( plot )->plant->harvest_multiplier;
-                    plant_count = std::min( std::max( plant_count, 1 ), 12 );
-
-                    // Multiply by the plant's and seed's base charges to mimic creating
-                    // items similar to iexamine::harvest_plant
-                    number_plants += plant_count * tmp.charges;
-                    number_seeds += std::max( 1, rng( plant_count / 4, plant_count / 2 ) ) * item_seed.charges;
-
-                    bay.i_clear( plot );
-                    bay.furn_set( plot, f_null );
-                    bay.ter_set( plot, t_dirtmound );
+                    results.splice( results.begin(), iexamine::harvest_items( *seed->type, bay, plot, skill_level ) );
                 }
             }
         }
     }
     bay.save();
     tmp = item( plant_types[plant_index], calendar::turn );
+
+    int number_plants = 0;
+    int number_seeds = 0;
+    for( std::list<item>::iterator i = results.begin(); i != results.end(); i++ ) {
+        if( ( *i ).typeId() == plant_types[plant_index] ) {
+            number_plants += i->count_by_charges() ? i->charges : 1;
+            i = results.erase( i );
+            i--;
+        } else if( ( *i ).typeId() == seed_types[plant_index] ) {
+            number_seeds += i->count_by_charges() ? i->charges : 1;
+            i = results.erase( i );
+            i--;
+        }
+    }
```

Comment:
I'd eventually like to count each item, not just the primary plant and seed, and then give more options when [forced] selling things.